cylc/flow/task_pool.py

```diff
@@ -822,7 +822,7 @@ def reload_taskdefs(self) -> None:
                 itask.copy_to_reload_successor(new_task)
                 self._swap_out(new_task)
                 LOG.info(f"[{itask}] reloaded task definition")
-                if itask.state(*TASK_STATUSES_ACTIVE):
+                if itask.state(*TASK_STATUSES_ACTIVE, TASK_STATUS_PREPARING):
```

Comment:
We should warn for preparing tasks as pre-reload settings may have been baked into the filesystem?